### ISSUE TITLE
Error when using with typing-extensions v4.1.1

### ISSUE THREAD

**Issue by DManowitz** [pyapp-kit/psygnal#220]
* psygnal version: 0.9.1
* Python version: 3.7/3.8
* Operating System: Windows 10

### Description

Got an error when using psygnal in an environment with typing-extensions 4.1.1:
```
File "C:\...\lib\site-packages\psygnal\_evented_model.py", line 94, in <module>
    @dataclass_transform(kw_only_default=True, field_specifiers=(Field, FieldInfo))
TypeError: dataclass_transform() got an unexpected keyword argument 'field_specifiers'
```

However, in another environment with typing-extensions v4.7.1, it ran without error.

I think this package needs to add a minimum version dependency for `typing-extensions`

### What I Did

```
Paste the command(s) you ran and the output.
If there was a crash, please include the traceback here.
```

**Comment by tlambert03** [comment]
For quite a while I avoided pinning typing extensions because of an annoying issue with tensor flow pinning an old version of typing extensions, it's a long story... 😅

I'll see if I can accomplish this by guarding the import in a type checking block while retaining the same functionality.  And if not will pin.


Thanks for letting me know :)

**Comment by tlambert03** [comment]
fix coming in #221 (not pinning typing extensions to 4.7.1, but making sure we can run all the way back to `typing-extensions==3.7.4.2`
